ios/Toodle/Rust/ToodleLib.swift

```diff
@@ -54,39 +54,63 @@ class ToodleLib {
         return OpaquePointer(AutoreleasingUnsafeMutablePointer<[OpaquePointer]>(&pointerArray))
     }
 
-    func allItems() -> [Item] {
-        let items = toodle_get_all_items(self.raw)
-        var allItems: [Item] = []
-        for index in 0..<item_list_count(items) {
-            let item = Item(raw: item_list_entry_at(items, Int(index))!)
-            allItems.append(item)
+    func allItems(completion: @escaping ([Item]) -> ()) {
+        let allItemsSQL = """
+                            [:find ?eid ?uuid ?name
+                            :where
+                            [?eid :todo/uuid ?uuid]
+                            [?eid :todo/name ?name]]
+                          """
+        do {
+            try self.query(sql: allItemsSQL).execute { (result) in
+                switch result {
+                    case .success(let rows):
+                        var items = [Item]()
+                        guard let rows = rows else {
+                            return
+                        }
+                        for row in rows {
+                            guard let id = row.asEntid(index: 0),
+                                let uuid = row.asUUID(index: 1),
+                                let name = row.asString(index: 2) else {
+                                    return
+                            }
+                            let item = Item(id: id, uuid: uuid, name: name)
+                            items.append(item)
+                        }
+                        completion(items)
```

Comment:
AIUI you only need a completion callback here because you're not using `Deferred`, right?